controllers/datadogagent/feature/kubernetesstatecore/feature.go

```diff
@@ -79,10 +79,15 @@ func (f *ksmFeature) Configure(dda *v2alpha1.DatadogAgent) feature.RequiredCompo
 		}
 	}
 
-	return feature.RequiredComponents{
-		ClusterAgent:       feature.RequiredComponent{IsRequired: &f.enable},
-		ClusterCheckRunner: feature.RequiredComponent{IsRequired: &f.clusterChecksEnabled},
+	output := feature.RequiredComponents{}
+	if f.enable {
```

Comment:
could we nest this (as well as the `if` block starting on line 72) within the first `if` block?